### PR TITLE
Api/setpass

### PR DESCRIPTION
**Pull request by MoritzHahn1337**
Added password change functionality with new API Endpoint

### CODE REVIEW

File: datameta.compose.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@ services:
       SQLALCHEMY_URL: postgresql://datameta:datameta@datameta-postgresql/datameta
       SESSION_URL: datameta-memcached:11211
       SESSION_KEY: datameta
-      SESSION_SECRET: # Set a 64 character random string here
+      SESSION_SECRET:
```
Comment (lkuchenb) [inline datameta.compose.yml]: Did you remove the comment intentionally?

File: datameta/static/js/account.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
+/*
+# Copyright (c) 2021 Universität Tübingen, Germany
+# Authors: Leon Kuchenbecker <leon.kuchenbecker@uni-tuebingen.de>
+# Authors: Moritz Hahn <moritz.hahn@uni-tuebingen.de>
```
Comment (lkuchenb) [inline datameta/static/js/account.js:4]: Let's just indent subsequent authors

```suggestion
#          Moritz Hahn <moritz.hahn@uni-tuebingen.de>
```

File: datameta/templates/account.pt [hunk 3]
```diff
@@ -1,7 +1,81 @@
...
+                            </span>
+                            <input type="password" class="form-control" name="new_password_repeat" id="new_password_repeat" placeholder="Repeat new Password">
+                            <div class="invalid-feedback">
+                                The two passwords do not match.
+                            </div>
+                        </div>
+
+                        <input type="hidden" id="uuid" name="uuid" value="${user.uuid}">
+                        <div class="input-group mb-4">
+                            <button class="btn btn-outline-danger" type="submit" id="submit_password">Submit Changes</button>
```
Comment (lkuchenb) [inline datameta/templates/account.pt:67]:
```suggestion
                            <button class="btn btn-outline-success" type="submit" id="submit_password">Submit Changes</button>
```

File: datameta/views/setpass.py [hunk 4]
```diff
@@ -31,29 +31,30 @@
...
-
-    # Validate password
-    if len(new_password) < 10:
-        raise HTTPBadRequest(json_body={'reason' : 'password too short'})
-
-    # Set the new password
-    user.pwhash = security.hash_password(new_password)
-
-    return HTTPNoContent()
+# Old API Endpoint
```
Comment (lkuchenb) [inline datameta/views/setpass.py:34]: Please remove the file entirely and also the delete the `add_route` call in `routes.py`

File: datameta/static/js/account.js [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+                    view_success();
+                    return;
+                } else {
+                    if (response.status == "401") {
+                        show_alert("You have to be logged in to perform this action.");
+                    } else if (response.status == "403") {
+                        show_alert("Wrong password.");
+                    } else if(response.status == "400") {
+                        show_alert("Your password has to have at least 10 Characters");   
+                    }
```
Comment (lkuchenb) [inline datameta/static/js/account.js:99]: Please add a fallback `else` clause with the same unknown error message as below

**Review by lkuchenb — CHANGES_REQUESTED** [review]
Thank you! I've added a few comments.

File: datameta.compose.yml (hunk 1 above)
Comment (MoritzHahn1337) [reply datameta.compose.yml]: Was already fixed in a30c1b8

File: datameta/static/js/account.js (hunk 5 above)
Comment (MoritzHahn1337) [reply datameta/static/js/account.js:99]: done

File: datameta/views/setpass.py (hunk 4 above)
Comment (MoritzHahn1337) [reply datameta/views/setpass.py:34]: Done

File: datameta/views/setpass.py [hunk 6]
```diff
@@ -1,107 +0,0 @@
-# Copyright (c) 2021 Universität Tübingen, Germany
```
Comment (lkuchenb) [inline datameta/views/setpass.py:1]: Please restore this file, this is the `/setpass` view for use with recovery tokens 😅 . Let's refactor that view to the new API endpoint in a separate PR.

**Review by lkuchenb — CHANGES_REQUESTED** [review]

File: datameta/views/setpass.py (hunk 6 above)
Comment (MoritzHahn1337) [reply datameta/views/setpass.py]: Done

**Review by lkuchenb — APPROVED** [review]
Sorry, I didn't realize the endpoints were defined in the same file at first. Looks good now!

**Pull request merged**